js/repeater.js

```diff
@@ -550,42 +606,35 @@
 					} else {
 						this.$nextBtn.focus();
 					}
-
+				} else if (this.$prevBtn.is(':disabled')) {
+					// if you can't focus, go the other way
+					this.$nextBtn.focus();
 				} else {
-					if (this.$prevBtn.is(':disabled')) {
-						// if you can't focus, go the other way
-						this.$nextBtn.focus();
-					} else {
-						this.$prevBtn.focus();
-					}
-
+					this.$prevBtn.focus();
 				}
-
 			}
 		},
 
-		previous: function () {
-			var d = 'disabled';
-			this.$nextBtn.attr(d, d);
-			this.$prevBtn.attr(d, d);
+		previous: function previous () {
+			this.$nextBtn.attr('disabled', 'disabled');
+			this.$prevBtn.attr('disabled', 'disabled');
 			this.pageIncrement = -1;
 			this.$element.trigger('previousClicked.fu.repeater');
 			this.render({
 				pageIncrement: this.pageIncrement
 			});
 		},
 
-		render: function (options) {
-			var self = this;
+		// This functions more as a "pre-render" than a true "render"
+		render: function render (opts) {
+			this.disable();
```

Comment:
`this.disable()` could theoretically get pushed further and further down as people add things to the code if they aren't paying attention. I put it at the very top, both as a signal to future developers that "the control is disabled while this function runs" and to prevent it getting accidentally pushed down too far (somehow) or lost entirely (very unlikely). I'm kind of treating it like an `e.preventDefault()` (which I would also put at the very top).